### PR TITLE
chore: parse and write the diff of `map` that are items in a `seq`

### PR DESCRIPTION
**Pull request by Lou1415926**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/template/diff/diff.go [hunk 1]
```diff
@@ -11,6 +11,11 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
+// For a property whose value is a list of map, map its name to the identifying property of the items.
+var mapItemsIdentifiedBy = map[string]string{
+	"ContainerDefinitions": "Name",
```
Comment (dannyrandall) [inline internal/pkg/template/diff/diff.go:16]: niceeeeee

File: internal/pkg/template/diff/diff_test.go [hunk 2]
```diff
@@ -336,6 +363,13 @@ Dog: (pleased) "ikr"`, t),
 	}
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
+			ogItemsIdentifiedBy := mapItemsIdentifiedBy
+			defer func() {
+				mapItemsIdentifiedBy = ogItemsIdentifiedBy
+			}()
+			mapItemsIdentifiedBy = map[string]string{
+				"StrawberryPopularitySurvey": "Name",
```
Comment (dannyrandall) [inline internal/pkg/template/diff/diff_test.go:371]: so random lol :joy:

File: internal/pkg/template/diff/diff.go (hunk 1 above)
Comment (dannyrandall) [reply internal/pkg/template/diff/diff.go:16]: though i wonder (nit) if something like `seqObjectIdentifiers` indicates better that it's only used to identify sequence objects? i don't feel too strongly abt it

**Review by dannyrandall — APPROVED** [review]
for the survey: i like strawberries

**Comment by Lou1415926** [comment]
Closing this PR for now. Rendering "implicit key" is just a bonus. The `(x unchanged items)` line in the diff is already enough for users to identify the items that have changed.


This would be a nice to have though. I will(might) reopen the PR in the future if I find a better implementation solution.

**Pull request closed without merging**